gammapy/irf/tests/test_background.py

```diff
@@ -49,3 +54,66 @@ def test_background_3d_write(bkg_3d):
     hdu = table_to_fits_table(bkg_3d.to_table())
     assert_equal(hdu.data['DETX_LO'][0], bkg_3d.data.axis('detx').lo.value)
     assert hdu.header['TUNIT1'] == bkg_3d.data.axis('detx').lo.unit
+
+
+def make_test_array():
+    # Create a dummy `Background2D`
+    # Make Axes
+    nE_bins = 100
+    ebounds = EnergyBounds.equal_log_spacing(0.1, 100, 100, 'TeV')
+    offset = Angle(np.linspace(0, 2.5, 100), "deg")
+    data = np.zeros((nE_bins, len(offset) - 1))
+    # Make dummy data
+    # At offset=0.59343434 deg and energy=0.11885022 TeV
+    data[2, 23] = 1
+    # At offset=1.50252525 deg and energy= 22.64644308 TeV
+    data[78, 59] = 1
+    data_unit = u.Unit('s-1 MeV-1 sr-1')
+    bkg_2d = Background2D(energy_lo=ebounds[:-1], energy_hi=ebounds[1:],
+                          offset_lo=offset[:-1], offset_hi=offset[1:],
+                          data=data * data_unit)
+    return bkg_2d
+
+
+def test_background2d_read_write(tmpdir):
+    bkg_2d_1 = make_test_array()
+    filename = str(tmpdir / "bkg2d.fits")
+    prim_hdu = fits.PrimaryHDU()
+    hdu_bkg = bkg_2d_1.to_fits()
+    hdulist = fits.HDUList([prim_hdu, hdu_bkg])
+    hdulist.writeto(filename)
```

Comment:
@cdeil 
Still get the same issue that if I am not writing the file, with only using from_hdulist(), I lost the dimensions quantity... I checked for example for the effective area, they are writting the file and using the function read then.